packages/idempotency/src/persistence/PersistenceLayer.ts

```diff
@@ -1,21 +1,96 @@
 /* eslint-disable @typescript-eslint/no-empty-function */
+import { createHash, Hash } from 'crypto';
+import { IdempotencyRecordStatus } from 'types/IdempotencyRecordStatus';
+import { EnvironmentVariablesService } from '../EnvironmentVariablesService';
 import { IdempotencyRecord } from './IdempotencyRecord';
 import { PersistenceLayerInterface } from './PersistenceLayerInterface';
 
 abstract class PersistenceLayer implements PersistenceLayerInterface {
-  public constructor() { }
-  public configure(_functionName: string = ''): void {}
+
+  // envVarsService is always initialized in the constructor
+  private envVarsService!: EnvironmentVariablesService;
+
+  private expiresAfterSeconds: number;
+
+  private functionName: string = '';
+
+  private hashFunction: Hash;
+
+  public constructor() { 
+    this.setEnvVarsService();
+    this.expiresAfterSeconds = 60 * 60; //one hour is the default expiration
+    this.hashFunction = createHash('md5');
+        
+  }
+  public configure(functionName: string = ''): void {
+    this.functionName = this.getEnvVarsService().getLambdaFunctionName + '.' + functionName;
+  }
+
   public async deleteRecord(): Promise<void> { }
   public async getRecord(): Promise<IdempotencyRecord> {
     return Promise.resolve({} as IdempotencyRecord);
   }
-  public async saveInProgress(): Promise<void> { }
+
+  /**
+   * Saves a record indicating that the function's execution is currently in progress
+   * 
+   * @param data - the data payload that will be hashed to create the hash portion of the idempotency key
+   */
+  public async saveInProgress(data: unknown): Promise<void> { 
```

Comment:
Perhaps this is out of scope, but is there a use case where 'data' would be undefined, such as if there is no input to the lambda? If so, are we comfortable with how the hash logic will respond?